dot/types/grandpa.go

```diff
@@ -111,19 +111,32 @@ func (v *GrandpaVoter) Decode(r io.Reader) error {
 		return err
 	}
 
-	v.Key = key
+	v.Key = *key
 	v.ID = id
 	return nil
 }
 
+// Encode will encode the GrandpaVoter
+func (v *GrandpaVoter) Encode() ([]byte, error) {
```

Comment:
nit: can you update all the receiver functions.
```suggestion
func (gv *GrandpaVoter) Encode() ([]byte, error) {
```